src/transformers/models/pop2piano/tokenization_pop2piano.py

```diff
@@ -0,0 +1,816 @@
+    to: this superclass for more information regarding those methods. However the code does not allow that and only
+    supports composing from various genres.
+
+    Args:
+        vocab_time_file (`str`):
+            Path to the tokenizer file which contains vocabulary for "TIME" tokens.
+        vocab_note_file (`str`):
+            Path to the tokenizer file which contains vocabulary for "NOTE" tokens.
+        vocab_velocity_file (`str`):
+            Path to the tokenizer file which contains vocabulary for "VELOCITY" notes.
+        vocab_special_file (`str`):
+            Path to the tokenizer file which contains vocabulary for "SPECIAL" tokens.
+        vocab_size_special (`int`, *optional*, defaults to 4):
+            Number of special values.
+        vocab_size_note (`int`, *optional*, defaults to 128):
+            Number of MIDI note tokens. Only the 88 pitches corresponding to piano keys are actually used.
+        vocab_size_velocity (`int`, *optional*, defaults to 2):
+            Number of velocity tokens.
+        vocab_size_time (`int`, *optional*, defaults to 100):
+            Number of beat shift tokens. Beat shifts indicate the relative time shift within the segment quantized into
+            8th-notes (half-beats).
+        default_velocity (`int`, *optional*, defaults to 77):
+            Determines the default velocity to use used while creating midi Notes.
+        num_bars (`int`, *optional*, defaults to 2):
+            Determines cutoff_time_idx in for each token.
+    """
+
+    model_input_names = ["token_ids"]
+    vocab_files_names = VOCAB_FILES_NAMES
+    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+
+    def __init__(
+        self,
+        vocab_time_file,
+        vocab_note_file,
+        vocab_velocity_file,
+        vocab_special_file,
+        vocab_size_time=100,
+        vocab_size_note=128,
+        vocab_size_special=4,
+        vocab_size_velocity=2,
+        default_velocity=77,
+        num_bars=2,
+        unk_token="-1",
+        eos_token="1",
+        pad_token="0",
+        bos_token="2",
+        **kwargs,
+    ):
+        unk_token = AddedToken(unk_token, lstrip=False, rstrip=False) if isinstance(unk_token, str) else unk_token
+        eos_token = AddedToken(eos_token, lstrip=False, rstrip=False) if isinstance(eos_token, str) else eos_token
+        pad_token = AddedToken(pad_token, lstrip=False, rstrip=False) if isinstance(pad_token, str) else pad_token
+        bos_token = AddedToken(bos_token, lstrip=False, rstrip=False) if isinstance(bos_token, str) else bos_token
+
+        super().__init__(
+            unk_token=unk_token,
+            eos_token=eos_token,
+            pad_token=pad_token,
+            bos_token=bos_token,
+            **kwargs,
+        )
+
+        self.vocab_size_time = vocab_size_time
+        self.vocab_size_note = vocab_size_note
+        self.vocab_size_special = vocab_size_special
+        self.vocab_size_velocity = vocab_size_velocity
```

Comment:
to remove in favor of the property that computes the length of the `self.token_special_encoder, self.token_time_encoder ,  self.token_note_encoder, self.token_velocity_encoder`

Reply:
we don't need those properties since we are not using them anywhere in the code(as they were mostly used in the complex logic of `_convert_token_to_id` and `_convert_id_to_token` and now the logic is replaced with `self.encoder.get`). So removing them altogether.